packages/lodestar/src/chain/chain.ts

```diff
@@ -94,15 +95,17 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
 
   public async getHeadStateContext(): Promise<ITreeStateContext> {
     //head state should always exist
-    return (await this.db.stateCache.get(this.forkChoice.headStateRoot()));
+    const headStateRoot = await this.db.stateCache.get(this.forkChoice.headStateRoot());
+    if (!headStateRoot) throw Error("headStateRoot does not exist");
```

Comment:
This check is potentially going to have a greater impact. Since "*head state should always exist*", is throwing an error the best solution?

Reply:
yup